File: P003/main.py
import time
import math

start = time.time()

def primeFactors(n):
    while n%2==0:
        print "2 "
        n /= 2

    for i in range(3, int(math.sqrt(n))):
        while n%i == 0:
            print str(i) + " "
            n/=i

    if n>2:
        print n



primeFactors(600851475143)

elapsed = time.time()

print("found in %s seconds") % (elapsed)

File: P003/main.java
import java.lang.*;

public class main {
    public static void primeFactors(long n) {
        while(n%2==0) {
            System.out.print("2 ");
            n/=2;
        }

        for(long i=3; i<=(int)Math.sqrt(n); i++) {
            while(n%i==0) {
                System.out.print(i + " ");
                n/=i;
            }
        }

        if(n>2) {
            System.out.print(n);
        }
        System.out.print("\n");
    }

    public static void main(String[] args) {
        long startTime = System.currentTimeMillis();


        primeFactors(600851475143L);

        long endTime = System.currentTimeMillis();
        long totalSpan = System.currentTimeMillis();
        System.out.println("Completed in " + totalSpan + " milliseconds.");
    }
}
